File: front/scada-front/src/app/DTOs/DigitalInputDTO.ts
export interface DigitalInputDTO {
    Name: string;
    Description: string;
    Address: string;
    ScanTime: number;
    IsScanning: boolean;
  }
  
  export function createDigitalInputDTO(Name: string, Description: string, Address: string, ScanTime: number, IsScanning: boolean): DigitalInputDTO {
    return {Name, Description, Address, ScanTime, IsScanning};
  }
  
  export interface DigitalInputIdDTO {
    Id: number;
    Name: string;
    Description: string;
    Address: string;
    ScanTime: number;
    IsScanning: boolean;
  }
  
  export function createDigitalInputIdDTO(Id: number, Name: string, Description: string, Address: string, ScanTime: number, IsScanning: boolean): DigitalInputIdDTO {
    return {Id, Name, Description, Address, ScanTime, IsScanning};
  }

File: back/scada-back/DTOs/DigitalInputDTO.cs
namespace scada_back.DTOs
{
    public class DigitalInputDTO
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public string Function { get; set; }

        public string Address { get; set; }

        public int ScanTime { get; set; }

        public bool IsScanning { get; set; }
    }
}
